### PR TITLE
Types refactor

### PR DESCRIPTION
**Pull request by igorrCarvalho**
This pull request aims to perform a comprehensive type refactor across the project, enhancing type safety and improving code clarity. The changes introduced in this PR will lead to better maintainability, reduced bugs related to type-related issues, and increased overall code quality.

### CODE REVIEW

**Review by lucaseduoli — CHANGES_REQUESTED** [review]
Need to merge the DEV changes with the types refactor. Putting JSX.Element as the component types is not a good practice. Take FloatComponent as an example: you need to create a specific type to each of the components.

**Review by anovazzi1 — CHANGES_REQUESTED** [review]

File: src/frontend/src/modals/baseModal/index.tsx
Comment (anovazzi1) [inline src/frontend/src/modals/baseModal/index.tsx:1]: improve the name of the type, it's the the header of the modal, so headerConst is not a good name

File: src/frontend/src/pages/FlowPage/components/PageComponent/index.tsx [hunk 1]
```diff
@@ -351,7 +351,7 @@ export default function Page({ flow }: { flow: FlowType }) {
     }
   }, [selectionEnded, lastSelection]);
 
-  const onSelectionChange = useCallback((flow) => {
+  const onSelectionChange = useCallback((flow): void => {
```
Comment (anovazzi1) [inline src/frontend/src/pages/FlowPage/components/PageComponent/index.tsx:354]: flow has any type here

File: src/frontend/src/types/components/index.ts [hunk 2]
```diff
@@ -169,6 +171,188 @@ export type IconComponentProps = {
...
+
+export type chatTriggerPropType = {
+  open: boolean;
+  isBuilt: boolean;
+  canOpen: boolean;
+  setOpen: (can: boolean) => void;
+};
+
+export type headerFlowsType = {
+  data: object;
```
Comment (anovazzi1) [inline src/frontend/src/types/components/index.ts:243]: object is not a good type practice, avoid using it

File: src/frontend/src/types/flow/index.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ export type NodeDataType = {
   type: string;
   node?: APIClassType;
   id: string;
-  value: any;
+  value: null;
```
Comment (anovazzi1) [inline src/frontend/src/types/flow/index.ts:21]: value is always null?

File: src/frontend/src/types/typesContext/index.ts
Comment (anovazzi1) [inline src/frontend/src/types/typesContext/index.ts:1]: avoid using any as type

**Review by anovazzi1 — CHANGES_REQUESTED** [review]

**Pull request merged**